src/test/java/com/maxmind/geoip2/model/JsonTest.java

```diff
@@ -1,112 +1,304 @@
-            + "       \"is_satellite_provider\" : true,\n"
-            + "       \"autonomous_system_organization\" : \"AS Organization\",\n"
-            + "       \"user_type\" : \"college\",\n"
-            + "       \"organization\" : \"Blorg\",\n"
-            + "       \"domain\" : \"example.com\"\n"
-            + "    },\n"
-            + "    \"country\" : {\n"
-            + "       \"names\" : {\n"
-            + "          \"en\" : \"United States of America\"\n"
-            + "       },\n"
-            + "       \"geoname_id\" : 1,\n"
-            + "       \"iso_code\" : \"US\",\n"
-            + "       \"confidence\" : 99\n"
-            + "    },\n"
-            + "    \"continent\" : {\n"
-            + "       \"names\" : {\n"
-            + "          \"en\" : \"North America\"\n"
-            + "       },\n"
-            + "       \"code\" : \"NA\",\n"
-            + "       \"geoname_id\" : 42\n"
-            + "    },\n"
-            + "    \"location\" : {\n"
-            + "       \"average_income\" : 24626,\n"
-            + "       \"population_density\" : 1341,\n"
-            + "       \"time_zone\" : \"America/Chicago\",\n"
-            + "       \"accuracy_radius\" : 1500,\n"
-            + "       \"metro_code\" : 765,\n"
-            + "       \"latitude\" : 44.98,\n"
-            + "       \"longitude\" : 93.2636\n"
-            + "    },\n"
-            + "    \"subdivisions\" : [\n"
-            + "       {\n"
-            + "          \"confidence\" : 88,\n"
-            + "          \"iso_code\" : \"MN\",\n"
-            + "          \"geoname_id\" : 574635,\n"
-            + "          \"names\" : {\n"
-            + "             \"en\" : \"Minnesota\"\n"
-            + "          }\n"
-            + "       },\n"
-            + "       {\n"
-            + "          \"iso_code\" : \"TT\"\n"
-            + "       }\n"
-            + "    ],\n"
-            + "    \"represented_country\" : {\n"
-            + "       \"geoname_id\" : 3,\n"
-            + "       \"names\" : {\n"
-            + "          \"en\" : \"United Kingdom\"\n"
-            + "       },\n"
-            + "       \"type\" : \"C<military>\",\n"
-            + "       \"iso_code\" : \"GB\"\n"
-            + "    },\n"
-            + "    \"postal\" : {\n"
-            + "       \"code\" : \"55401\",\n"
-            + "       \"confidence\" : 33\n"
-            + "    },\n"
-            + "    \"city\" : {\n"
-            + "       \"confidence\" : 76,\n"
-            + "       \"geoname_id\" : 9876,\n"
-            + "       \"names\" : {\n"
-            + "          \"en\" : \"Minneapolis\"\n"
-            + "       }\n"
-            + "    }\n"
-            + " }\n";
+    @Test
+    public void testInsightsSerialization() throws IOException {
+        String json = JSON.std
+                .composeString()
+                .startObject()
+                .startObjectField("maxmind")
```

Comment:
This is so much nicer than the string concat, and less error prone. Wrong nesting of startObject*/end is as easy to achieve here as the old version, but that is nothing in comparison.